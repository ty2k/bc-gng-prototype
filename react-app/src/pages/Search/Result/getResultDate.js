import getDate from "../../../_utils/getDate";

// Return a result's date in YYYY-MM-DD format
export default function getResultDate(result) {
  let date = "";

  // Prefer the <FS name="date" /> node if available
  if (result?.FS?.length > 0) {
    result?.FS?.forEach((detail) => {
      if (detail?.$?.NAME === "date") {
        date = detail?.$?.VALUE;
      }
    })
  }

  // Or try <MT N="DCTERMS.modified" />
  if (!date && result?.MT?.length > 0) {
    result?.MT?.forEach((metaTag) => {
      if (metaTag?.$?.N === "DCTERMS.modified") {
        date = metaTag?.$?.V;
      }
    })
  }

  // Or try <MT N="datasource/modificationDate" />
  if (!date && result?.MT?.length > 0) {
    result?.MT?.forEach((metaTag) => {
      if (metaTag?.$?.N === "datasource/modificationDate") {
        date = getDate.iso8601Extended(parseInt(metaTag?.$?.V));
      }
    })
  }

  // Or try <MT N="mes:date" />
  if (!date && result?.MT?.length > 0) {
    result?.MT?.forEach((metaTag) => {
      if (metaTag?.$?.N === "mes:date") {
        date = getDate.iso8601Extended(parseInt(metaTag?.$?.V));
      }
    })
  }

  if (date) {
    return <p className="date"><strong>Published date:</strong> {date}</p>;
  }

  return null;
}
