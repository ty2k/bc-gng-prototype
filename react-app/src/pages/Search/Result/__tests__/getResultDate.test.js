import TestRenderer from "react-test-renderer";

import getResultDate from "../getResultDate";

import {
  mockResultTab1,
  mockResultTab2,
  mockResultTab3,
  mockResultTab4,
} from "../__mocks__/mockResultData";

describe("getResultDate", () => {
  const dateTab1 = TestRenderer.create(
    getResultDate(mockResultTab1)
  ).toJSON();
  const dateTab2 = TestRenderer.create(
    getResultDate(mockResultTab2)
  ).toJSON();
  const dateTab3 = TestRenderer.create(
    getResultDate(mockResultTab3)
  ).toJSON();
  const dateTab4 = TestRenderer.create(
    getResultDate(mockResultTab4)
  ).toJSON();

  it("renders a Tab 1 (All) date correctly", () => {
    expect(dateTab1).toMatchSnapshot();
  });
  it("renders a Tab 2 (News) date correctly", () => {
    expect(dateTab2).toMatchSnapshot();
  });
  it("renders a Tab 3 (Documents) date correctly", () => {
    expect(dateTab3).toMatchSnapshot();
  });
  it("renders a Tab 4 (Services) date correctly", () => {
    expect(dateTab4).toMatchSnapshot();
  });
})
