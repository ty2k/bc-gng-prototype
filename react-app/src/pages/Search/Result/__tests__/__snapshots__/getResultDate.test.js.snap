// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getResultDate renders a Tab 1 (All) date correctly 1`] = `
<p
  className="date"
>
  <strong>
    Published date:
  </strong>
   
  2018-12-18
</p>
`;

exports[`getResultDate renders a Tab 2 (News) date correctly 1`] = `
<p
  className="date"
>
  <strong>
    Published date:
  </strong>
   
  2019-08-01
</p>
`;

exports[`getResultDate renders a Tab 3 (Documents) date correctly 1`] = `
<p
  className="date"
>
  <strong>
    Published date:
  </strong>
   
  2021-03-24
</p>
`;

exports[`getResultDate renders a Tab 4 (Services) date correctly 1`] = `
<p
  className="date"
>
  <strong>
    Published date:
  </strong>
   
  2020-03-31
</p>
`;
